src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -6,18 +6,50 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int [] diagonal = new int[matrix[0].length];
+        for (int i = 0; i < matrix[0].length; i++) {
+            diagonal[i] = matrix[i][i];
+        }
+        return diagonal;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int [] counterDiagonal = new int[matrix[0].length];
+        int counter = matrix[0].length - 1;
+        for (int i = 0; i < matrix[0].length; i++) {
+            counterDiagonal[i] = matrix[i][counter];
+            counter--;
+        }
+        return counterDiagonal;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        int min = 0;
+        if (matrix.length == 1) {
+            return matrix[0][0];
+        }
+        for (int i = 0; i < matrix[0].length; i++) {
+            for (int j = 0; j < matrix[1].length; j++) {
+                if (matrix[i][j] < min) {
+                    min = matrix[i][j];
+                }
+            }
+        }
+        return min;
     }
 
     public int getMax(int[][] matrix) {
-        return 0;
+        int max = 0;
```

Comment:
what if we would we have the matrix only with negative numbers?

Reply:
Fixed this issue